spotlight-client/src/NarrativeLayout/NarrativeNavigation/Wayfinder.tsx

```diff
@@ -0,0 +1,301 @@
+    padding: ${rem(24)} ${rem(16)};
+  }
+`;
+
+const activeButtonStyles = css`
+  background: ${palette.signal.links} !important;
+  color: ${palette.white} !important;
+`;
+
+const flashingButtonStyles = css`
+  animation: blinker 1s infinite;
+
+  @keyframes blinker {
+    0% {
+      opacity: 0.3;
+    }
+    50% {
+      opacity: 1;
+    }
+    100% {
+      opacity: 0.3;
+    }
+  }
+`;
+
+export const Button = styled(BasicButton)<{
+  active?: boolean;
+  rounded?: boolean;
+  flashing?: boolean;
+}>`
+  min-width: ${(props) => rem(props.rounded ? 42 : 109)};
+  height: ${rem(42)};
+  background: ${palette.marble3} !important;
+  color: ${palette.slate85};
+  font-weight: 500;
+  font-size: ${rem(16)};
+  gap: ${rem(16)};
+  padding: 0;
+
+  &:hover:not([disabled]) {
+    ${activeButtonStyles}
+  }
+  &:disabled {
+    color: #c4c4c4;
+  }
+
+  ${(props) => props.active && activeButtonStyles}
+  ${(props) => props.flashing && flashingButtonStyles}
+`;
+
+const SectionTitle = styled(NarrativeSectionTitle)<{
+  minSize: number;
+  maxSize: number;
+}>`
+  margin: 0;
+  ${(props) => fluidFontSizeStyles(props.minSize, props.maxSize)}
+`;
+
+const IntroducingText = styled.div`
+  margin-bottom: ${rem(24)};
+  letter-spacing: -0.01em;
+  line-height: 130%;
+`;
+
+type NavigationProps = SectionNavProps & {
+  sections: LayoutSection[];
+  isOpen?: boolean;
+};
+
+const Wayfinder: React.FC<NavigationProps> = ({
+  isOpen = true,
+  activeSection,
+  sections,
+  goToSection,
+}) => {
+  const isMobile = useBreakpoint(false, ["mobile-", true]);
+
+  // used for Wayfinder first-run experience
+  // this tells whether user has visited page or not, depending on its  local store
+  const [hasVisitedPageBefore, setVisited] = useState(
+    localStorage.getItem("isVisited") === "hasVisited" || false
+  );
+  const [isOpenModal, openModal] = useState(false);
+  const [isOpenWayfinder, openWayfinder] = useState(isOpen);
+  const [isExpanded, setExpanded] = useState(false);
+  const [scrolled, setScrolled] = useState(0);
+
+  const visitPage = () => {
+    localStorage.setItem("isVisited", "hasVisited");
+    setVisited(true);
+  };
+
+  useEffect(() => {
+    if (!hasVisitedPageBefore) {
+      openModal(true);
+      visitPage();
+    }
+
+    if (scrolled > 98) {
+      openWayfinder(false);
+    } else {
+      openWayfinder(true);
+    }
+  }, [hasVisitedPageBefore, scrolled]);
+
+  const handlers = useSwipeable({
+    onSwipedUp: () => setExpanded(true),
+    onSwipedDown: () => setExpanded(false),
+    swipeDuration: 300,
+    preventScrollOnSwipe: true,
+  });
+
+  const totalPages = sections.length;
+  const disablePrev = activeSection === 1;
+  const disableNext = activeSection === totalPages;
+
+  return createPortal(
+    <>
+      <Modal
+        isOpen={isOpenModal}
+        onRequestClose={() => openModal(false)}
+        shouldCloseOnOverlayClick={false}
+      >
+        <ModalHeading>
+          Introducing <strong>Wayfinder 2.1</strong>
```

Comment:
So we want to tell our users "Introducing Wayfinder 2.1"? What is Wayfinder, should users know about this?

Reply:
It's more like a placeholder. Final copy is up to @hobuobi